lib/rspec/matchers/built_in/yield.rb

```diff
@@ -326,7 +337,7 @@ def negative_failure_reason
           end
```

Comment:
Am I right in thinking that it shouldn't be calling `all_args_match?` either, and should instead be checking whether the args matched when yielded?